src/Runner/Runner.php

```diff
@@ -89,26 +124,229 @@ public function __construct(
+                    // File analysis result (we want close-to-realtime progress with frequent cache savings)
+                    if (ParallelAction::WORKER_RESULT === $workerResponse['action']) {
+                        $fileAbsolutePath = $workerResponse['file'];
+                        $fileRelativePath = $this->directory->getRelativePathTo($fileAbsolutePath);
+
+                        // Dispatch an event for each file processed and dispatch its status (required for progress output)
+                        $this->dispatchEvent(
+                            FixerFileProcessedEvent::NAME,
+                            new FixerFileProcessedEvent($workerResponse['status'])
+                        );
+
+                        if (isset($workerResponse['fileHash'])) {
+                            $this->cacheManager->setFileHash($fileRelativePath, $workerResponse['fileHash']);
+                        }
+
+                        foreach ($workerResponse['errors'] ?? [] as $error) {
+                            $this->errorsManager->report(new Error(
+                                $error['type'],
+                                $error['filePath'],
+                                null !== $error['source']
+                                    ? SourceExceptionFactory::fromArray($error['source'])
+                                    : null,
+                                $error['appliedFixers'],
+                                $error['diff']
+                            ));
+                        }
+
+                        // Pass-back information about applied changes (only if there are any)
+                        if (isset($workerResponse['fixInfo'])) {
+                            $changed[$fileRelativePath] = $workerResponse['fixInfo'];
+
+                            if ($this->stopOnViolation) {
+                                $processPool->endAll();
+
+                                return;
+                            }
+                        }
+
+                        return;
+                    }
+
+                    if (ParallelAction::WORKER_GET_FILE_CHUNK === $workerResponse['action']) {
+                        // Request another chunk of files, if still available
+                        $fileChunk = $getFileChunk();
+
+                        if (0 === \count($fileChunk)) {
+                            $process->request(['action' => ParallelAction::RUNNER_THANK_YOU]);
+                            $processPool->endProcessIfKnown($identifier);
+
+                            return;
+                        }
+
+                        $process->request(['action' => ParallelAction::RUNNER_REQUEST_ANALYSIS, 'files' => $fileChunk]);
+
+                        return;
+                    }
+
+                    if (ParallelAction::WORKER_ERROR_REPORT === $workerResponse['action']) {
+                        throw WorkerException::fromRaw($workerResponse); // @phpstan-ignore-line
```

Comment:
side note:
yes, we pass `action` to `fromRaw()`. if we would have `{action: string, payload: datamodel}`, we would be able to pass data model only ;P